Service/release_action_service.go

```
@@ -4,100 +4,162 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
 
 	"github.com/actions-on-packages/gh-action-release/internal"
+	"github.com/actions-on-packages/gh-action-release/logger"
 	models "github.com/actions-on-packages/gh-action-release/types"
-	gh "github.com/cli/go-gh"
-	"github.com/cli/go-gh/pkg/api"
 	ghRepo "github.com/cli/go-gh/pkg/repository"
 )
 
 type ActionReleaseService interface {
 	CreateRelease(repo ghRepo.Repository, options models.CreateReleaseApiRequest) (*models.ReleaseApiResponse, error)
-	IsRepoAction(current_repo ghRepo.Repository) bool
+	IsActionRepo(current_repo ghRepo.Repository) bool
+	GetExistingReleases(repo ghRepo.Repository) ([]models.ReleaseApiResponse, error)
+	GetLatestRelease(repo ghRepo.Repository) (models.ReleaseApiResponse, error)
+	MigrateActionRelease(repo ghRepo.Repository, options models.MigrationOptions) bool
+	UploadBlob(fileName string, repo ghRepo.Repository, options models.MigrationOptions) error
 }
 
-func GetExistingReleases(repo ghRepo.Repository) []models.ReleaseApiResponse {
-	opts := api.ClientOptions{
-		Host: repo.Host(),
+func GetActionPackagesTagList(repo ghRepo.Repository) (models.PackagesTagListResponse, error) {
+	var tagListResponse models.PackagesTagListResponse
+	restClient, err := internal.GetRestClient(repo)
+	if err != nil {
+		return tagListResponse, err
+	}
+	var clientOptions = fmt.Sprintf("%+v", restClient)
+	// logger.LogInfo(clientOptions)
+	options := strings.Split(clientOptions, "Authorization:token ")[1]
+	authToken := strings.Split(options, " ")[0]
+
+	var tagListUrl = fmt.Sprintf("https://ghcr.io/v2/%s/%s/tags/list", repo.Owner(), repo.Name())
+	req, err := http.NewRequest("GET", tagListUrl, nil)
+	req.Header.Add("Authorization", "bearer "+internal.EncodeToBase64(authToken))
+	response, err := http.DefaultClient.Do(req)
+	if response.StatusCode != 200 {
+		logger.LogInfo("unable to retrieve tag list from GHCR ")
```

LogError in place of logInfo. Also not to ignore the error message from the response

---

done